### PR TITLE
support pageable query

### PR DESCRIPTION
**Pull request by sophiaso**

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java [hunk 1]
```diff
@@ -396,7 +399,7 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
     private <T> boolean isCrossPartitionQuery(@NonNull DocumentQuery query, @NonNull Class<T> domainClass) {
         final Optional<String> partitionKeyName = getPartitionKeyField(domainClass);
 
-        if (!partitionKeyName.isPresent()) {
+        if (!partitionKeyName.isPresent() || query == null) {
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: looks we can remove `query == null` if there is `ALL` type.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentDbPageRequest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        if (!(obj instanceof DocumentDbPageRequest)) {
+            return false;
+        }
+
+        final DocumentDbPageRequest that = (DocumentDbPageRequest) obj;
+
+        return super.equals(that) && (requestContinuation != null ?
+                requestContinuation.equals(that.requestContinuation) : that.requestContinuation == null);
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentDbPageRequest.java]: I think the expression is a little complicated.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/DocumentDbQueryExecution.java [hunk 3]
```diff
@@ -52,4 +55,24 @@ public Object execute(DocumentQuery query, Class<?> type, String collection) {
...
+        private final Pageable pageable;
+
+        public PagedExecution(DocumentDbOperations operations, Pageable pageable) {
+            this.operations = operations;
+            this.pageable = pageable;
+        }
+
+        @Override
+        public Object execute(DocumentQuery query, Class<?> type, String collection) {
+            if (pageable.getPageNumber() != 0 &&  !(pageable instanceof DocumentDbPageRequest)) {
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/DocumentDbQueryExecution.java]: two space after `&&`

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java (hunk 1 above)
Comment (sophiaso) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java]: Will update in another PR.

**Pull request merged**